cmd/streaming-signature-v4.go

```diff
@@ -347,9 +358,14 @@ func (cr *s3ChunkedReader) Read(buf []byte) (n int, err error) {
 		cr.err = io.EOF
 		return n, cr.err
 	}
-
-	cr.offset = copy(buf, cr.buffer)
-	n += cr.offset
+	// since we have choosen internal buf
+	if internalBuf {
+		cr.offset = copy(buf, choosenBuf)
+		n += cr.offset
+	} else {
+		// looks like we had enough buf given by the caller to read.
+		n += len(buf)
```

Comment:
Your read is most likely bigger than the size read.
```suggestion
		n += int64(size)
```